### PR TITLE
downgrade lease integration for teams

### PR DESCRIPTION
**Pull request by maxtaco**
- figure out who is being downgraded
- take out leases on all of those users

### CODE REVIEW

**Comment by maxtaco** [comment]
This won't work until we land keybase/keybase#1380

File: go/teams/teams.go [hunk 1]
```diff
@@ -263,52 +263,97 @@ func (t *Team) applicationKeyForMask(mask keybase1.ReaderKeyMask, secret []byte)
...
+	return res, nil
+}
+
+func (t *Team) getDowngradedUsers(ms *memberSet) (users []string, err error) {
+
+	for _, member := range ms.None {
+		users = append(users, member.version.Username)
+	}
+
+	for _, member := range ms.Writers {
```
Comment (oconnor663) [inline go/teams/teams.go]: What if an admin/owner is downgraded to a reader? Are we catching that case?

Comment (maxtaco) [inline go/teams/teams.go]: no, good catch.  i wonder why i ignored that

**Comment by maxtaco** [comment]
Can I collapse all 3 of these PRs into the bottom one, and then rebase that one, and fix this bug?

File: go/teams/teams.go (hunk 1 above)
Comment (maxtaco) [inline go/teams/teams.go]: ok, fixed back in #7336

**Comment by maxtaco** [comment]
thanks for the review

**Pull request merged**